training/src/main/scala/com/airbnb/aerosolve/training/DecisionTreeTrainer.scala

```diff
@@ -72,6 +72,15 @@ object DecisionTreeTrainer {
         .filter(x => x.contains(rankKey))
         .take(candidateSize)
 
+    val ex = examples(0)
+    val numFeatures = LinearRankerUtils.getNumFeatures(ex, rankKey)
```

Comment:
For the pricing model, different examples always have different number of features which is a small subset of the total number of features we have, for example `San Francisco` might be a missing feature for a listing in `Tokyo`. Due to this reason, I realized there is already randomness in the way we are choosing feature split candidates here. But I guess this change is useful for cases where all examples have a full set of features.